lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1599,3 +1600,47 @@ def invalidate_generated_certificates(course_id, enrolled_students, certificate_
         download_url='',
         grade='',
     )
+
+
+def upload_ora2_data(
+        _xmodule_instance_args, _entry_id, course_id, _task_input, action_name
+):
+    """
+    Collect ora2 responses and upload them to S3 as a CSV
+    """
+
+    start_date = datetime.now(UTC)
+    start_time = time()
+
+    num_attempted = 1
+    num_total = 1
+
+    task_progress = TaskProgress(action_name, num_total, start_time)
+    task_progress.attempted = num_attempted
+
+    curr_step = {'step': "Collecting responses"}
+
+    task_progress.update_task_state(extra_meta=curr_step)
+
+    try:
+        header, datarows = OraAggregateData.collect_ora2_data(course_id)
+        rows = [header] + [row for row in datarows]
+    # Update progress to failed regardless of error type
+    except Exception:  # pylint: disable=broad-except
+        TASK_LOG.exception('Failed to get ORA data.')
+        task_progress.failed = 1
+        curr_step = {'step': "Error while collecting data"}
+
+        task_progress.update_task_state(extra_meta=curr_step)
+
+        return UPDATE_STATUS_FAILED
+
+    task_progress.succeeded = 1
+    curr_step = {'step': "Uploading CSV"}
+    task_progress.update_task_state(extra_meta=curr_step)
+
+    upload_csv_to_report_store(rows, 'ORA_data', course_id, start_date)
+
+    task_progress.update_task_state(extra_meta=curr_step)
```

Comment:
1. I wasn't sure if they were expecting those specific values from this particular task because they wrote tests around them (SUCCEEDED and FAILED), so I left them in this way.
2. This is slightly different from the other tasks in that most of the row-by-row computation was happening elsewhere.